### DISCUSSION TITLE
Which source codes are used to customize titlebar?

### DISCUSSION THREAD

**Discussion by AkiraVoid** [microsoft/WinUI-Gallery#1114] [Q&A] [answered]
I noticed that Gallery's titlebar is fully customized and looks awesome, I want to achieve similar effects in my own apps. I didn't find the source codes which is used to customize titlebar in this repo, can you please give me a link or something?

**Comment by marcelwgn** [comment, marked as answer]
The custom TitleBar is mainly done by extending the content into the TitleBar:
https://github.com/microsoft/WinUI-Gallery/blob/4d3d763fcddfcaf07215385b3133073039f7e086/WinUIGallery/Helper/NavigationOrientationHelper.cs#L59-L60

The displayed Titlebar is created here:

https://github.com/microsoft/WinUI-Gallery/blob/4d3d763fcddfcaf07215385b3133073039f7e086/WinUIGallery/Navigation/NavigationRootPage.xaml#L55-L66

**Reply by AkiraVoid** [reply]
Excuse me, how you made the back button of NavigationView appears in the title bar?

**Reply by AkiraVoid** [reply]
Oh I found out, it's a little bit like absolute position in CSS, I didn't know that 2 elements in same grid column will be placed together, not like StackPanel does.

**Reply by marcelwgn** [reply]
Sorry for the late reply. Yes Grid places items on top of each other if they are in the same grid column and row, this is why we are able to show the Navigation back button and the title of the app next to each other. Is there anything else I can help with or would you mind marking this question as answered for other users?

**Reply by AkiraVoid** [reply]
That's all, thanks for your help.
